docs/02 - Contributing/G - End to End Testing with Different Configs.md

```diff
@@ -0,0 +1,89 @@
+# End-to-End Testing with Different Configurations
+
+End-to-end testing of different Anchor Platform configurations can be accomplished with `docker-compose`. We will
+maintain a base file (`integration-tests/docker-compose-configs/docker-compose.base.yaml`) and pass in an override
+file (`integration-tests/docker-compose-configs/*/docker-compose.override.yaml`) to specify which config files to use
+and what `end_to_end_tests` to run against the Anchor Platform configuration. 
+
+end_to_end_tests.py is a Python CLI tool used to test end-to-end **Anchor Platform + Anchor Server + Kafka + 
+Horizon** data flows.
+
+Please refer to [End to End Tests](/end-to-end-tests/README.md) for information on the end-to-end tests.
+
+
+### Anchor Platform Configuration Files:
+Each **directory** in `integration-tests/docker-compose-configs` contains all the configuration files 
+necessary for a full Anchor Platform deployment.
+1) `anchor-platform-config` - the configuration file for **Anchor Platform**
+2) `anchor-reference-server-config.yaml` - the configuration file for the **Anchor Reference Server**
+3) `assets-test.json` - the assets file to be used by the **Anchor Platform**
+4) `docker-compose-config.override.yaml` - this file gets merged into `docker-compose-configs/docker-compose.base.yaml
+when running docker-compose`. This file contains references to the specific files (the config directory) to be used
+and which tests (end-to-end) are to be run against the setup.
+5) `stellar.toml` - the TOML file thats served from Anchor Platform's `/.well-known/stellar.toml` endpoint
```

Comment:
There's a lot of code repetition across these directories. Can we have a strategy where we have base files and only override them?

I can't easily understand the difference between the `*.yaml`, `*.toml` or `*.json` files for each of these 3 cases. Maintaining this can become a hard and error-prone task because it's very difficult visually spot what changes among these options.

Reply:
thats a good point. i'll  move all the *yaml, *toml, *json files up one directory and use those as default config files for all tests when no override is provided.

Reply:
added in c6ef400